protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/DataCmdHandler.java

```diff
@@ -126,13 +137,19 @@ public void destroy() {
      *
      */
     public Response onCommand(SMTPSession session, Request request) {
-        String parameters = request.getArgument();
-        Response response = doDATAFilter(session,parameters);
-        
-        if (response == null) {
-            return doDATA(session, parameters);
-        } else {
-            return response;
+        TimeMetric timeMetric = metricFactory.timer("SMTP-" + request.getCommand());

```

Comment:
It's java 7

It means that you can not do it beautiful in smtp and imap. And you have to cope with the close() method signature, not that explicit...